processor/labelsprocessor/labels_processor.go

```diff
@@ -0,0 +1,124 @@
+			return fmt.Errorf("Labels Processor configuration contains duplicate keys")
+		}
+		keys[elem.Key] = true
+	}
+
+	return nil
+}
+
+func (lp *labelMetricProcessor) ProcessMetrics(_ context.Context, md pdata.Metrics) (pdata.Metrics, error) {
+
+	otlpMetrics := pdata.MetricsToOtlp(md)
+
+	for _, otlpMetric := range otlpMetrics {
+		for _, instrMetric := range otlpMetric.GetInstrumentationLibraryMetrics() {
+			for _, metric := range instrMetric.GetMetrics() {
+
+				// Multiple types of Data Points exists, and each of them must be handled differently
+				if metric.GetIntSum() != nil {
+					intDataPoints := metric.GetIntSum().GetDataPoints()
+					handleIntDataPoints(intDataPoints, lp)
+				} else if metric.GetIntGauge() != nil {
+					intDataPoints := metric.GetIntGauge().GetDataPoints()
+					handleIntDataPoints(intDataPoints, lp)
+				} else if metric.GetDoubleGauge() != nil {
+					doubleDataPoints := metric.GetDoubleGauge().GetDataPoints()
+					handleDoubleDataPoints(doubleDataPoints, lp)
+				} else if metric.GetDoubleSum() != nil {
+					doubleDataPoints := metric.GetDoubleSum().GetDataPoints()
+					handleDoubleDataPoints(doubleDataPoints, lp)
+				} else if metric.GetIntHistogram() != nil {
+					intHistogramDataPoints := metric.GetIntHistogram().GetDataPoints()
+					handleIntHistogramDataPoints(intHistogramDataPoints, lp)
+				} else if metric.GetDoubleHistogram() != nil {
+					doubleHistogramDataPoints := metric.GetDoubleHistogram().GetDataPoints()
+					handleDoubleHistogramDataPoints(doubleHistogramDataPoints, lp)
+				}
+
+			}
+		}
+	}
+
+	return md, nil
+}
+
+func handleIntDataPoints(intDataPoints []*v1.IntDataPoint, lp *labelMetricProcessor) {
+	for _, label := range lp.cfg.Labels {
+		for _, dataPoint := range intDataPoints {
+			deDuplicateAndAppend(&dataPoint.Labels, label.Key, label.Value)
+		}
+	}
+}
+
+func handleDoubleDataPoints(doubleDataPoints []*v1.DoubleDataPoint, lp *labelMetricProcessor) {
+	for _, label := range lp.cfg.Labels {
+		for _, dataPoint := range doubleDataPoints {
+			deDuplicateAndAppend(&dataPoint.Labels, label.Key, label.Value)
+		}
+	}
+}
+
+func handleIntHistogramDataPoints(intHistogramDataPoints []*v1.IntHistogramDataPoint, lp *labelMetricProcessor) {
+	for _, label := range lp.cfg.Labels {
+		for _, dataPoint := range intHistogramDataPoints {
+			deDuplicateAndAppend(&dataPoint.Labels, label.Key, label.Value)
+		}
+	}
+}
+
+func handleDoubleHistogramDataPoints(doubleHistogramDataPoints []*v1.DoubleHistogramDataPoint, lp *labelMetricProcessor) {
+	for _, label := range lp.cfg.Labels {
+		for _, dataPoint := range doubleHistogramDataPoints {
+			deDuplicateAndAppend(&dataPoint.Labels, label.Key, label.Value)
+		}
+	}
+}
+
+// This processor will always by default update existing label values. Also assumes duplicate labels do not already exist in the metric
+func deDuplicateAndAppend(labels *[]*v11.StringKeyValue, key string, value string) {
+	// If the key already exists, overwrite it
```

Comment:
nit: Usually comment like this is not useful as we can see what you are doing from the code quite easily. Also comment like this may start to be a lie as time goes; when other started to modify code but not the comment. 

If you really think that comment like this is needed, then consider restructuring your code like the following to use method name as comment (but I don't think it's needed in your case, so following is just example).

```go
if label, ok := findLabelByName(labels, key); ok {
   overrideExistingLabel(labels, value);
} else {
  appendNewLabel(labels, key, valye)
}

```

This applies to the `// If it does not exist, append it` comment too.

Reply:
Makes sense to me, comments will be removes